internal/metrics/collectors/manager.go

```diff
@@ -88,12 +109,76 @@ func (nc *LocalManagerMetricsCollector) UpdateLastReloadTime(duration time.Durat
 	nc.lastReloadTime.Set(float64(duration / time.Millisecond))
 }
 
+// UpdateWorkerProcessCount sets the number of NGINX worker processes
+func (nc *LocalManagerMetricsCollector) UpdateWorkerProcessCount(configVersion string) {
+	totalWorkerProcesses, err := getWorkerProcesses()
+	if err != nil {
+		glog.Errorf("unable to collect process metrics : %v", err)
+		return
+	}
+	nc.workerProcessTotal.WithLabelValues(configVersion).Set(totalWorkerProcesses)
+	nc.workerProcessTotal.WithLabelValues("old").Set(nc.oldWorkerProcessTotal)
+	nc.oldWorkerProcessTotal = totalWorkerProcesses
+}
+
+func getWorkerProcesses() (float64, error) {
+	var workerProcesses float64
+	procFolders, err := ioutil.ReadDir("/proc")
+	if err != nil {
+		return 0, fmt.Errorf("unable to read directory /proc : %v", err)
+	}
+	masterPidFile, err := os.Open("/var/lib/nginx/nginx.pid")
+	if err != nil {
+		return 0, fmt.Errorf("unable to get master PID : %v", err)
+	}
+	defer masterPidFile.Close()
+
+	var masterPid string
+	masterPidScanner := bufio.NewScanner(masterPidFile)
+	for masterPidScanner.Scan() {
+		masterPid = masterPidScanner.Text()
+	}
+	if err := masterPidScanner.Err(); err != nil {
+		return 0, fmt.Errorf("could not read /var/lib/nginx/ngin.pid : %v", err)
+	}
+
+	for _, f := range procFolders {
+		u, err := user.LookupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Uid))
+		if err != nil {
+			return 0, fmt.Errorf("could not find user ID for file %v : %v", f.Name(), err)
+		}
+		if u.Name == "nginx user" {
+			statusFile := fmt.Sprintf("/proc/%v/status", f.Name())
+			f, err := os.Open(statusFile)
+			if err != nil {
+				return 0, fmt.Errorf("unable to open file %v", statusFile)
+			}
+			defer f.Close()
+
+			scanner := bufio.NewScanner(f)
+
+			for scanner.Scan() {
+				if strings.HasPrefix(scanner.Text(), "PPid") {
+					ppidLineSplit := strings.Split(scanner.Text(), "\t")
+					ppid := ppidLineSplit[len(ppidLineSplit)-1]
+
+					if ppid != masterPid {
```

Comment:
In the case of a worker process: The `ppid` of the process should **match** the pid of the master process.
```suggestion
					if ppid == masterPid {
```